CHANGELOG.md

```diff
@@ -4,6 +4,96 @@ All notable changes to Router will be documented in this file.
+### Rename `originating_request` to `supergraph_request` on various plugin `Request` structures ([Issue #1713](https://github.com/apollographql/router/issues/1713))
+
+We feel that `supergraph_request` makes it more clear that this is the request received from the client.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/1715
+
+### Prometheus is no longer defaulting to the GraphQL endpoint and listener address ([Issue #1645](https://github.com/apollographql/router/issues/1645))
+
+The Router's Prometheus interface is now exposed at `127.0.0.1:9090/metrics`, rather than `http://0.0.0.0:4000/plugins/apollo.telemetry/prometheus`.  This should be both more secure and also more generally compatible with the default settings that Prometheus expects (which also uses port `9090` and just `/metrics` as its defaults).
+
+To expose to a non-localhost interface, it is necessary to explicitly opt-into binding to a socket address of `0.0.0.0:9090` (i.e., all interfaces on port 9090) or a specific available interface (e.g., `192.168.4.1`) on the host.
+
+Have a look at the _Features_ section (below) to learn how to customize the listen address and the path.
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/1654
+
+## 🚀 Features
+
+### New plugin helper: `map_first_graphql_response` ([Issue #1564](https://github.com/apollographql/router/issues/1564))
+
+In supergraph and execution services, the service response contains not just one GraphQL response but a stream of them, in order to support features such as `@defer`.
+
+This new method of `ServiceExt` and `ServiceBuilderExt` in `apollo_router::layers` wraps a service and calls a `callback` when the first GraphQL response in the stream returned by the inner service becomes available.  The callback can then access the HTTP parts (headers, status code, etc) or the first GraphQL response before returning them.
+
+See the doc-comments in `apollo-router/src/layers/mod.rs` for more.
+
+By [@SimonSapin](https://github.com/SimonSapin) in https://github.com/apollographql/router/pull/1708
+
+### Users can customize the Prometheus listener address and URL path ([Issue #1645](https://github.com/apollographql/router/issues/1645))
+
+You can now customize the Prometheus listener socket address and URL path in your YAML configuration:
+
+```yaml
+telemetry:
+  metrics:
+    prometheus:
+      listen: 127.0.0.1:9090 # default
+      path: /metrics # default
+      enabled: true
+```
+
+By [@o0Ignition0o](https://github.com/@o0Ignition0o) in https://github.com/apollographql/router/pull/1654
+
+### Add an `apollo_router::graphql::ResponseStream` type alias ([PR #1697](https://github.com/apollographql/router/pull/1697))
+
+It is equivalent to `BoxStream<'static, graphql::Response>` and makes
+some type signatures slightly simpler.
+
+By [@SimonSapin](https://github.com/SimonSapin) in https://github.com/apollographql/router/pull/1697
+
```

Comment:
I think this is a breaking change.

Reply:
Adding a type alias sounds like a feature to me — is it possible that the breaking part is actually above in Breaking Changes already?

Reply:
It's ok. I wanted to make sure that:
                    let response: http::Response<BoxStream<'static, graphql::Response>> = res.response;
                    let response: http::Response<graphql::ResponseStream> = res.response;
were equivalent. I just compiled that code and they are, so it's not a breaking change.